### DISCUSSION TITLE
Payjoin

### DISCUSSION THREAD

**Discussion by ConorOkus** [cashubtc/nutshell#543] [Ideas]
In Payjoin the recipient of a transaction appends an input and outputs to a transaction to disrupt the "common input heuristic" and gain more privacy.  

Does Payjoin make sense on the inbound UTXOs when depositing to a mint?

https://payjoin.org/

**Comment by thesimplekid** [comment]
I think it does. The payjoin endpoint can be expose by the mint either in the information of the on-chain mint https://github.com/cashubtc/nuts/pull/107 or to be defined separately in another optional nut, likely better as the later.  I'm interested in adding support for it in CDK using `rust-payjoin`.
